Composer/packages/client/src/pages/publish/Publish.tsx

```diff
@@ -3,357 +3,222 @@
-        botStatus.status = history.status;
-      }
+      tmpBot.publishTarget = publishTargets[0].name;
     }
-    statusList.push(botStatus);
+    botList.push(tmpBot);
   });
-  return { botSettingList, statusList, botPublishTypesList };
+  return { botSettingList, botPublishTargetsList, botPublishTypesList, botList };
 };
+
 const Publish: React.FC<RouteComponentProps<{ projectId: string; targetName?: string }>> = (props) => {
   const { projectId = '' } = props;
   const botProjectData = useRecoilValue(localBotsDataSelector);
   const publishHistoryList = useRecoilValue(localBotPublishHistorySelector);
   const {
     getPublishHistory,
-    getPublishStatus,
-    getPublishTargetTypes,
+    getPublishStatusV2,
     setPublishTargets,
     publishToTarget,
     setQnASettings,
-    rollbackToVersion: rollbackToVersionDispatcher,
     addNotification,
     deleteNotification,
   } = useRecoilValue(dispatcherState);
+  // fill Settings, status, publishType, publish target for bot from botProjectMeta
+  const { botSettingList, botPublishTargetsList, botPublishTypesList, botList } = useMemo(() => {
+    return generateComputedData(botProjectData);
+  }, [botProjectData]);
 
-  const [selectedBots, setSelectedBots] = useState<IBotStatus[]>([]);
-  const [publishDisabled, setPublishDisabled] = useState(false);
-
-  const [showNotifications, setShowNotifications] = useState<Record<string, boolean>>({});
-  const [currentBotPublishTargetList, setCurrentBotPublishTargetList] = useState<
-    { projectId: string; publishTarget: string }[]
-  >([]);
-  // fill Settings, status, publishType, publish target for bot from botProjectMeta, publishHistory
-  const { botSettingList, statusList, botPublishTypesList } = useMemo(() => {
-    return generateComputedData(botProjectData, publishHistoryList, currentBotPublishTargetList);
-  }, [botProjectData, publishHistoryList]);
+  const showNotificationsRef = useRef<Record<string, boolean>>({});
 
-  const [botStatusList, setBotStatusList] = useState<IBotStatus[]>(statusList);
-  const [botPublishHistoryList, setBotPublishHistoryList] = useState<
-    { projectId: string; publishHistory: { [key: string]: PublishResult[] } }[]
-  >(publishHistoryList);
-  const [showLog, setShowLog] = useState(false);
-  const [publishDialogHidden, setPublishDialogHidden] = useState(true);
-  const [pullDialogHidden, setPullDialogHidden] = useState(true);
-
-  const [showAuthDialog, setShowAuthDialog] = useState(false);
-
-  // items to show in the list
-  const [selectedVersion, setSelectedVersion] = useState<PublishResult | null>(null);
+  useEffect(() => {
+    botList
+      .filter((bot) => !!bot.publishTarget && !pollingUpdaterList.some((u) => u.beEqual(bot.id, bot.publishTarget)))
+      .map((bot) => {
+        const updater = new PublishStatusPollingUpdater({ botProjectId: bot.id, targetName: bot.publishTarget });
+        updater.start(updateData);
+        pollingUpdaterList.push(updater);
```

Comment:
Could you explain this section. I'm slighltly lost. On every useEffect you create a new pollingUpdater object. Why do we do this?

Reply:
There is a `pollingUpdater` List, so we create a new updater after filtering the existing updater. 
`
      .filter(
        (bot) => !!bot.publishTarget && !pollingUpdaterList.some((u) => u.isSameUpdater(bot.id, bot.publishTarget))
      )
`

Reply:
Can we connect offline on this. Adding, unit test cases testing the scenario would also be helpful in understanding this flow.